packages/@aws-cdk/aws-kinesisfirehose/test/integ.delivery-stream.source-stream.ts

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env node
+import * as iam from '@aws-cdk/aws-iam';
+import * as kinesis from '@aws-cdk/aws-kinesis';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as cdk from '@aws-cdk/core';
+import * as constructs from 'constructs';
+import * as firehose from '../lib';
+
+const app = new cdk.App();
+
+const stack = new cdk.Stack(app, 'aws-cdk-firehose-delivery-stream-source-stream');
+
+const bucket = new s3.Bucket(stack, 'Bucket', {
+  removalPolicy: cdk.RemovalPolicy.DESTROY,
+});
+
+const role = new iam.Role(stack, 'Role', {
+  assumedBy: new iam.ServicePrincipal('firehose.amazonaws.com'),
+});
+
+const mockS3Destination: firehose.IDestination = {
+  bind(_scope: constructs.Construct, _options: firehose.DestinationBindOptions): firehose.DestinationConfig {
+    const bucketGrant = bucket.grantReadWrite(role);
+    return {
+      extendedS3DestinationConfiguration: {
+        bucketArn: bucket.bucketArn,
+        roleArn: role.roleArn,
+      },
+      dependables: [bucketGrant],
+    };
+  },
+};
```

Comment:
Is this duplicated with the other integration test? Can you make one shared construct for this for the unit and integ tests?

Reply:
I think for literacy purposes, I'd rather keep this duplicated